### ISSUE TITLE
Add rule checking for `if __name__ ==  "__main__":` clause in test files

### ISSUE THREAD

**Issue by TobiasDijkhuis** [astral-sh/ruff#27926]
### Summary

```python
import pytest

def test_addition():
   assert 1 + 1 == 2

if __name__ == "__main__":
    pytest.main()
```
It is in my opinion much nicer to just have
```python
def test_addition():
   assert 1 + 1 == 2
```
and invoke the tests as (or whatever the file is called)
```pytest test_addition.py```

Even if you disagree with this, calling `pytest.main` does not raise a `SystemExit`, and only returns the return code (see documentation [here](https://docs.pytest.org/en/stable/how-to/usage.html#calling-pytest-from-python-code)). Thus the code in the first example I showed does not give an error (i.e. would pass in CI), even if the test fails.
A milder alternative could be to detect 
```python
if __name__ == "__main__":
    pytest.main()
```
which should then be corrected to 
```python
if __name__ == "__main__":
    sys.exit(pytest.main())
```
to properly fail if any of the tests fail.
